content/en/docs/examples/microservices-istio/add-new-microservice-version/index.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Add a new version of reviews
+overview: Deploy a new version of a microservice.
+
+weight: 50
+
+---
+
+In this module, you deploy a new version of the `reviews` microservice, _v2_,
+which will return the number and star color of ratings provided by reviewers. In
+a real-world scenario, you would perform lint tests, unit tests, integration
+tests, end-to-end tests and tests in a staging environment, before the
+deployment.
+
+1.  Deploy the new version of the `reviews` microservice without the
+    `app=reviews` label. Without that label, the new version will not be
+    selected to provide the `reviews` service. As such, it will not be called by
+    the production code. Run the following command to deploy the `reviews`
+    microservice version 2, while replacing the label `app=reviews` by
+    `app=reviews_test`:
+
+    {{< text bash >}}
+    $ curl -s {{< github_file >}}/samples/bookinfo/platform/kube/bookinfo.yaml | sed 's/app: reviews/app: reviews_test/' | kubectl apply -l app=reviews_test,version=v2 -f -
+    deployment "reviews-v2" created
+    {{< /text >}}
+
+1.  Access your application to ensure the deployed microservice did not disrupt
+    it.
+
+1.  Test the new version of your microservice from inside the cluster using the
+    testing container you deployed earlier. Note that your new version accesses
+    the production pods of the `ratings` microservice during the test. Also note
+    that you have to use the pod IP to access your new version of the
+    microservice, because it is not selected for the `reviews` service.
+
+    1.  Get the IP of the pod:
+
+        {{< text bash >}}
+        $ REVIEWS_V2_POD_IP=$(kubectl get pod -l app=reviews_test,version=v2 -o jsonpath='{.items[0].status.podIP}')
+        $ echo $REVIEWS_V2_POD_IP
+        {{< /text >}}
+
+    1.  Send a request to the pod and see that it returns the correct result:
+
+        {{< text bash >}}
+        $ kubectl exec -it $(kubectl get pod -l app=sleep -o jsonpath='{.items[0].metadata.name}') -- curl $REVIEWS_V2_POD_IP:9080/reviews/7
+        {"id": "7","reviews": [{  "reviewer": "Reviewer1",  "text": "An extremely entertaining play by Shakespeare. The slapstick humour is refreshing!", "rating": {"stars": 5, "color": "black"}},{  "reviewer": "Reviewer2",  "text": "Absolutely fun and entertaining. The play lacks thematic depth when compared to other plays by Shakespeare.", "rating": {"stars": 4, "color": "black"}}]}
+        {{< /text >}}
+
+    1.  Perform primitive load testing by sending a request 10 times in a row:
+
+        {{< text bash >}}
+        $ kubectl exec -it $(kubectl get pod -l app=sleep -o jsonpath='{.items[0].metadata.name}') -- sh -c "for i in 1 2 3 4 5 6 7 8 9 10; do curl -o /dev/null -s -w '%{http_code}\n' $REVIEWS_V2_POD_IP:9080/reviews/7; done"
```

Comment:
To avoid a full re-test of the tutorial, I'm limiting my changes in this PR to language only not code (unless the code is absolutely broken), but I added this issue to track the suggested improvement:
https://github.com/istio/istio.io/issues/6450